layer/compile_time_layer.cc

```diff
@@ -28,6 +29,12 @@ namespace {
 // Layer book-keeping information
 // ----------------------------------------------------------------------------
 
+using time_point = std::chrono::system_clock::time_point;
```

Comment:
Can you add a comment explaining why we want to use this particular clock instead `steady_clock` or `high_resolution_clock`?